fxa-client/src/lib.rs

```diff
@@ -294,6 +294,44 @@ impl FirefoxAccount {
         Ok(Some(self.handle_oauth_token_response(resp, None)?))
     }
 
+    pub fn begin_pairing_flow(&mut self, pairing_url: &str, scopes: &[&str]) -> Result<String> {
+        let mut url = self.state.config.content_url_path("/pair/supp")?;
+        let pairing_url = Url::parse(pairing_url)?;
```

Comment:
@eoger the approach here is to hide the details of the whole `fragment` thing into the rust lib, so we don't have to do it in Kotlin, Swift, etc..

Reply:
Please check that the `pairing_url` has the same origin as the content-serve from `self.state.config`.